File: Numerical_Methods_for_Physics_Garcia/edition2/Fortran/rand.f
      real*8 function rand( seed )
      integer*4 seed
! Random number generator; Uniform dist. in [0,1)
! Input
!   seed    Integer seed (DO NOT USE A SEED OF ZERO)
! Output
!   rand    Random number uniformly distributed in [0,1)

      static a,m
      real*8 a,m,temp
      data a,m/16807.0, 2147483647.0/

      temp = a*seed
      seed = dmod(temp,m)
      rand = seed/m

      return
      end

File: Numerical_Methods_for_Physics_Garcia/edition2/Cpp/rand.cpp
#include "NumMeth.h"

// Random number generator; Uniform dist. in [0,1)
double rand( long& seed ) {
// Input
//   seed    Integer seed (DO NOT USE A SEED OF ZERO)
// Output
//	 rand    Random number uniformly distributed in [0,1)

  const double a = 16807.0;
  const double m = 2147483647.0;
  double temp = a * seed;
  seed = (long)(fmod(temp,m));
  double rand = seed/m;
  return( rand );
}